### PR TITLE
Updating pyproject python versioning

### LINKED ISSUE #450: Bug: Constrain python versioning within the `pyproject` file to match with CI
### Example code with output

This is the error below trying to install pycytominer with python 3.12
```
Package operations: 6 installs, 1 update, 0 removals

  - Downgrading numpy (2.1.1 /home/conda/feedstock_root/build_artifacts/numpy_1725411817006/work/dist/numpy-2.1.1-cp312-cp312-linux_x86_64.whl -> 1.24.4): Failed

  ChefBuildError

  Backend 'setuptools.build_meta:__legacy__' is not available.
  
  Cannot import 'setuptools.build_meta'

  at ~/.local/share/pypoetry/venv/lib/python3.10/site-packages/poetry/installation/chef.py:164 in _prepare
      160│ 
      161│                 error = ChefBuildError("\n\n".join(message_parts))
      162│ 
      163│             if error is not None:
    → 164│                 raise error from None
      165│ 
      166│             return path
      167│ 
      168│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:
```

### Issue description

There is currently a minor issue where Pycytominer can be installed with Python 3.12, even though it hasn't been tested for compatibility. In the `pyproject.toml` and `poetry.lock` files, the Python version is set to `>=3.8`, which allows users with Python 3.12 to install Pycytominer, despite the lack of testing.

Integration testing only covers Python versions 3.8 to 3.11:
[Integration Test File](https://github.com/cytomining/pycytominer/blob/4b772a5d7e7ded6c5ae67a024f28102c4d2305db/.github/workflows/integration-test.yml#L64-L68)

`pyproject.toml` file allowing 3.12 to be installed:
[Pyproject File](https://github.com/cytomining/pycytominer/blob/4b772a5d7e7ded6c5ae67a024f28102c4d2305db/pyproject.toml#L45)

When attempting to use Python 3.12, an error occurs during the installation with Poetry (shown above).

It appears that a downgraded version of NumPy cannot be installed with Python 3.12, causing the installation to fail. This issue might need an adjustment to specify supported Python versions accurately or to test compatibility with Python 3.12.

### Expected behavior

Pycytominer should be restricted from being installed by users with Python 3.12.

```toml
[tool.poetry.dependencies]
python          = ">=3.8,<3.12" # making it more restrictive 
numpy           = ">=1.16.5"
scipy           = ">=1.5"
pandas          = ">=1.2.0"
scikit-learn    = ">=0.21.2"
sqlalchemy      = ">=1.3.6, <2"
pyarrow         = ">=8.0.0"

```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by axiomcura**
# Description

Thank you for your contribution to pycytominer!
Please _succinctly_ summarize your proposed change.
What motivated you to make this change?

This PR closes #450, where Python 3.12 faces compatibility issues with downgrading to NumPy 1.24. Additionally, Python 3.12 is currently not supported by Pycytominer, as there are no tests involving Pycytominer with Python 3.12. This change will prevent users with Python 3.12 from installing Pycytominer.

Here's the test log below:
```text
(pycytominer) (base) erikserrano@mahogany:~/Development/pycytominer$ poetry run pytest
=================================================================================== test session starts ===================================================================================
platform linux -- Python 3.11.10, pytest-8.1.1, pluggy-1.4.0
rootdir: /home/erikserrano/Development/pycytominer
configfile: pyproject.toml
testpaths: tests
plugins: cov-4.1.0
collected 158 items                                                                                                                                                                       

tests/test_aggregate.py ..........                                                                                                                                                  [  6%]
tests/test_annotate.py .......                                                                                                                                                      [ 10%]
tests/test_consensus.py ...                                                                                                                                                         [ 12%]
tests/test_cyto_utils/test_DeepProfiler_processing.py ....                                                                                                                          [ 15%]
tests/test_cyto_utils/test_annotate_custom.py ......                                                                                                                                [ 18%]
tests/test_cyto_utils/test_cell_locations.py ......                                                                                                                                 [ 22%]
tests/test_cyto_utils/test_cells.py .......s...............                                                                                                                         [ 37%]
tests/test_cyto_utils/test_collate.py ....                                                                                                                                          [ 39%]
tests/test_cyto_utils/test_cp_image_features.py ..                                                                                                                                  [ 41%]
tests/test_cyto_utils/test_feature_blocklist.py ..                                                                                                                                  [ 42%]
tests/test_cyto_utils/test_feature_drop_outlier.py ....                                                                                                                             [ 44%]
tests/test_cyto_utils/test_feature_infer.py .....                                                                                                                                   [ 48%]
tests/test_cyto_utils/test_features_count_na.py .                                                                                                                                   [ 48%]
tests/test_cyto_utils/test_features_util.py .                                                                                                                                       [ 49%]
tests/test_cyto_utils/test_load.py ......                                                                                                                                           [ 53%]
tests/test_cyto_utils/test_modz.py .......                                                                                                                                          [ 57%]
tests/test_cyto_utils/test_output.py .......                                                                                                                                        [ 62%]
tests/test_cyto_utils/test_single_cell_ingest_utils.py ...                                                                                                                          [ 63%]
tests/test_cyto_utils/test_util.py ................                                                                                                                                 [ 74%]
tests/test_cyto_utils/test_write_gct.py ....                                                                                                                                        [ 76%]
tests/test_feature_select.py ..........                                                                                                                                             [ 82%]
tests/test_normalize.py ............                                                                                                                                                [ 90%]
tests/test_operations/test_correlation_threshold.py ....                                                                                                                            [ 93%]
tests/test_operations/test_get_na_columns.py ...                                                                                                                                    [ 94%]
tests/test_operations/test_transform.py ....                                                                                                                                        [ 97%]
tests/test_operations/test_variance_threshold.py ....                                                                                                                               [100%]

==================================================================================== warnings summary =====================================================================================
tests/test_cyto_utils/test_cells.py::test_load_compartment
  /home/erikserrano/Development/pycytominer/pycytominer/cyto_utils/cells.py:423: RemovedIn20Warning: Deprecated API features detected! These feature(s) are not compatible with SQLAlchemy 2.0. To prevent incompatible upgrades prior to updating applications, ensure requirements files are pinned to "sqlalchemy<2.0". Set environment variable SQLALCHEMY_WARN_20=1 to show all deprecation warnings.  Set environment variable SQLALCHEMY_SILENCE_UBER_WARNING=1 to silence this message. (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
    (num_rows,) = next(self.conn.execute(f"SELECT COUNT(*) FROM {table}"))

tests/test_cyto_utils/test_collate.py::test_base_case
  /home/erikserrano/Programs/miniconda3/envs/pycytominer/lib/python3.11/site-packages/cytominer_database/utils.py:5: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources

tests/test_cyto_utils/test_collate.py::test_base_case
tests/test_cyto_utils/test_collate.py::test_base_case
tests/test_cyto_utils/test_collate.py::test_base_case
tests/test_cyto_utils/test_collate.py::test_base_case
  /home/erikserrano/Programs/miniconda3/envs/pycytominer/lib/python3.11/site-packages/pkg_resources/__init__.py:2825: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('sphinxcontrib')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

tests/test_cyto_utils/test_util.py::test_check_compartments
  /home/erikserrano/Development/pycytominer/pycytominer/cyto_utils/util.py:59: UserWarning: Non-canonical compartment detected: CeLLs
    warnings.warn(warn_str)

tests/test_cyto_utils/test_util.py::test_check_compartments
  /home/erikserrano/Development/pycytominer/pycytominer/cyto_utils/util.py:59: UserWarning: Non-canonical compartment detected: CeLLs, nucLEI, CYTOplasm
    warnings.warn(warn_str)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================= 157 passed, 1 skipped, 8 warnings in 555.20s (0:09:15) ==================================================================

```

Please also link to any relevant issues that your code is associated with.

## What is the nature of your change?

- [X] Bug fix (fixes an issue).
- [ ] Enhancement (adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [X] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have deleted all non-relevant text in this pull request template.

----
📚 Documentation preview 📚: https://pycytominer--454.org.readthedocs.build/en/454/

### CODE REVIEW

**Review by d33bs — APPROVED** [review]
LGTM - feel free to merge this in after tests pass!

Aside, if you'd like to automatically close an issue after a PR gets merged, you can use the keyword "closes", for example: `closes #<issue number>` (it will also display with a dotted line and hint when rendered by GitHub). Consider adding this to the PR description to help close out the issue once ready to merge.

**Comment by axiomcura** [comment]
> LGTM - feel free to merge this in after tests pass!
> 
> Aside, if you'd like to automatically close an issue after a PR gets merged, you can use the keyword "closes", for example: `closes #<issue number>` (it will also display with a dotted line and hint when rendered by GitHub). Consider adding this to the PR description to help close out the issue once ready to merge.

Oh I didn't know that! thanks! Now merging !

**Comment by axiomcura** [comment]
Thanks, @d33bs! I'll merge once the tests are complete. I had to resolve merge conflicts and update this branch accordingly.

**Pull request merged**
